### PR TITLE
User sign up

### PR DESCRIPTION
**Pull request by saurishkar**

### CODE REVIEW

**Comment by MatthewWaite** [comment]
This is great! Thanks for adding this feature, @saurishkar. When are you planning on merging?

File: src/apis/user.js [hunk 1]
```diff
@@ -6,7 +6,15 @@ const UserAPI = {
       .post(`${ process.env.REACT_APP_AMS_API_BASE }/users/token`)
       .set('Accept', 'application/json')
       .send(params);
+  },
+  signup: (params) => {
+  	return request
+  		.post('http://spree-ams.herokuapp.com/api/v1/users')
```
Comment (ShubhamGupta) [inline src/apis/user.js]: Do not hard-code the URL, tokens and anything else. Use `.env` file for the same.

File: src/components/shared/header.jsx [hunk 2]
```diff
@@ -18,10 +19,13 @@ class Header extends Component {
   constructor(props) {
     super(props);
 
-    this.state = { showModal: false };
+    this.state = { showModal: false, showSignUpModal: false };
```
Comment (ShubhamGupta) [inline src/components/shared/header.jsx]: Please rename old `showModal` to `showLoginModal`. Make sure the functionality is still running fine.

File: src/components/user-signup.jsx [hunk 3]
```diff
@@ -1,66 +1,136 @@
...
+                  </div>
+                </div>
+              </div>
+            </div>
+         </Modal>
     );
   }
 }
 
+// function validate(values) {
```
Comment (ShubhamGupta) [inline src/components/user-signup.jsx]: Please remove these comments.

File: src/components/user-signup.jsx [hunk 4]
```diff
@@ -1,66 +1,136 @@
...
+                            type="password" 
+                            component="input" />
+                          </div>
+                        </div>
+                      </div>
+                    </div>
+                    <br />
+                    <div className="form-group clearfix">
+                      <div className="col-sm-12 text-center">
+                          <button className="button-primary">
```
Comment (ShubhamGupta) [inline src/components/user-signup.jsx:95]: `left-align` this button. Just to keep it consistent with the submit button in login form modal.

File: src/containers/user-signup-connector.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { connect } from 'react-redux';
+
+import UserSignUp from '../components/user-signup';
+import Actions from '../actions';
+import UserAPI from '../apis/user';
+
+function mapDispatchToProps(dispatch) {
+	return {
```
Comment (ShubhamGupta) [inline src/containers/user-signup-connector.js:8]: Fix indentation.

**Review by ShubhamGupta — CHANGES_REQUESTED** [review]

**Review by ShubhamGupta — APPROVED** [review]

**Pull request merged**